{"url": "https://api.github.com/repos/rails-girls-summer-of-code/rgsoc-teams/pulls/comments/162044237", "pull_request_review_id": 89432051, "id": 162044237, "diff_hunk": "@@ -4,5 +4,12 @@ module HasSeason\n \n   included do\n     belongs_to :season\n+\n+    # @param season [#name, String, Integer]\n+    # @return [ActiveRecord::Relation]\n+    def self.in_season(season)\n+      season = season.try(:name) || season", "path": "app/models/concerns/has_season.rb", "position": 8, "original_position": 8, "commit_id": "bc2fe44cb300ce4481c00a998ecb38c4614294f6", "original_commit_id": "bc2fe44cb300ce4481c00a998ecb38c4614294f6", "user": {"login": "carpodaster", "id": 164400, "avatar_url": "https://avatars2.githubusercontent.com/u/164400?v=4", "gravatar_id": "", "url": "https://api.github.com/users/carpodaster", "html_url": "https://github.com/carpodaster", "followers_url": "https://api.github.com/users/carpodaster/followers", "following_url": "https://api.github.com/users/carpodaster/following{/other_user}", "gists_url": "https://api.github.com/users/carpodaster/gists{/gist_id}", "starred_url": "https://api.github.com/users/carpodaster/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/carpodaster/subscriptions", "organizations_url": "https://api.github.com/users/carpodaster/orgs", "repos_url": "https://api.github.com/users/carpodaster/repos", "events_url": "https://api.github.com/users/carpodaster/events{/privacy}", "received_events_url": "https://api.github.com/users/carpodaster/received_events", "type": "User", "site_admin": false}, "body": "The argument \"union type\" requires the `try`. There is a difference between `try` and `&`, they're not the same thing:\r\n\r\nObject#try is\r\n```ruby\r\ndef try(meth)\r\n  send(meth) if respond_to?(meth)\r\nend\r\n```\r\n**Edit**: `nil` check is implied as it responds to almost nothing\r\n\r\nThe safe navigation is:\r\n```\r\nsend(meth) unless nil?\r\n```\r\n\r\n```\r\n$ rails c\r\nLoading development environment (Rails 5.1.4)\r\n2.5.0 :001 > RUBY_VERSION\r\n => \"2.5.0\"\r\n2.5.0 :002 > \"\".try(:name)\r\n => nil\r\n2.5.0 :003 > nil.try(:name)\r\n => nil\r\n2.5.0 :004 > nil&.name\r\n => nil\r\n2.5.0 :005 > \"\"&.name\r\nTraceback (most recent call last):\r\nNoMethodError (undefined method `name' for \"\":String)\r\n```", "created_at": "2018-01-17T13:04:47Z", "updated_at": "2018-01-17T13:21:49Z", "html_url": "https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/910#discussion_r162044237", "pull_request_url": "https://api.github.com/repos/rails-girls-summer-of-code/rgsoc-teams/pulls/910", "author_association": "OWNER", "_links": {"self": {"href": "https://api.github.com/repos/rails-girls-summer-of-code/rgsoc-teams/pulls/comments/162044237"}, "html": {"href": "https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/910#discussion_r162044237"}, "pull_request": {"href": "https://api.github.com/repos/rails-girls-summer-of-code/rgsoc-teams/pulls/910"}}, "body_html": "<p>The argument \"union type\" requires the <code>try</code>. There is a difference between <code>try</code> and <code>&amp;</code>, they're not the same thing:</p>\n<p>Object#try is</p>\n<div class=\"highlight highlight-source-ruby\"><pre><span class=\"pl-k\">def</span> <span class=\"pl-en\">try</span>(<span class=\"pl-smi\">meth</span>)\n  send(meth) <span class=\"pl-k\">if</span> respond_to?(meth)\n<span class=\"pl-k\">end</span></pre></div>\n<p><strong>Edit</strong>: <code>nil</code> check is implied as it responds to almost nothing</p>\n<p>The safe navigation is:</p>\n<pre><code>send(meth) unless nil?\n</code></pre>\n<pre><code>$ rails c\nLoading development environment (Rails 5.1.4)\n2.5.0 :001 &gt; RUBY_VERSION\n =&gt; \"2.5.0\"\n2.5.0 :002 &gt; \"\".try(:name)\n =&gt; nil\n2.5.0 :003 &gt; nil.try(:name)\n =&gt; nil\n2.5.0 :004 &gt; nil&amp;.name\n =&gt; nil\n2.5.0 :005 &gt; \"\"&amp;.name\nTraceback (most recent call last):\nNoMethodError (undefined method `name' for \"\":String)\n</code></pre>", "body_text": "The argument \"union type\" requires the try. There is a difference between try and &, they're not the same thing:\nObject#try is\ndef try(meth)\n  send(meth) if respond_to?(meth)\nend\nEdit: nil check is implied as it responds to almost nothing\nThe safe navigation is:\nsend(meth) unless nil?\n\n$ rails c\nLoading development environment (Rails 5.1.4)\n2.5.0 :001 > RUBY_VERSION\n => \"2.5.0\"\n2.5.0 :002 > \"\".try(:name)\n => nil\n2.5.0 :003 > nil.try(:name)\n => nil\n2.5.0 :004 > nil&.name\n => nil\n2.5.0 :005 > \"\"&.name\nTraceback (most recent call last):\nNoMethodError (undefined method `name' for \"\":String)", "in_reply_to_id": 162040036}